src/PixiEditor/ViewModels/SubViewModels/Main/LayersViewModel.cs

```diff
@@ -48,10 +53,80 @@ public bool CanDeleteSelected()
     [Command.Basic("PixiEditor.Layer.DeleteSelected", "LAYER_DELETE_SELECTED", "LAYER_DELETE_SELECTED_DESCRIPTIVE", CanExecute = "PixiEditor.Layer.CanDeleteSelected", IconPath = "Trash.png")]
     public void DeleteSelected()
     {
-        var member = Owner.DocumentManagerSubViewModel.ActiveDocument?.SelectedStructureMember;
+        var doc = Owner.DocumentManagerSubViewModel?.ActiveDocument;
+
+        var member = doc.SelectedStructureMember;
         if (member is null)
             return;
+
         member.Document.Operations.DeleteStructureMember(member.GuidValue);
+        var allLayers = doc.StructureHelper.GetAllMembers().ToArray();
+
+        //Declare the layer that will be selected
+        StructureMemberViewModel baseLayer = member;
+
+        //Checking if the length is greater than one, to avoid crashing.
+        if (allLayers.Length > 0)
+        {  
+            for (int i = 0; i < allLayers.Length; i++)
+            {
+                //Finding the layer that is being deleted
+                if (allLayers[i] == member)
+                {
+                    //Selecting the new layer
+                    if (i > 0)
+                    {
+                        baseLayer = allLayers[i - 1];
+                    }
+                    else if (i == allLayers.Length - 1)
+                    {
+                        baseLayer = allLayers[i];
+                    }
+                    else if (i == 0)
+                    {
+                        baseLayer = allLayers[i + 1];
+                    }
+
+                    break;
+                }
+                else
+                {
+                    //Managing elements inside folders
```

Comment:
Unless I'm missing something here (I haven't run the code, sry) this doesn't handle nested folders (since folders can contain other folders that can contain other folders..). I think you need an algorithm that goes something like:
1. Find the parent of the member being deleted (`DocumentStructureModule.FindChildAndParentOrThrow`)
2. If the parent has only one child (which is getting deleted) then select the parent
3. Otherwise, use a loop to find the closest child to the one getting deleted (like you are already doing).
4. Select found child